vllm/executor/ray_gpu_executor.py

```diff
@@ -234,6 +243,19 @@ def _init_workers_ray(self, placement_group: "PlacementGroup",
                           max_concurrent_workers=self.parallel_config.
                           max_parallel_loading_workers)
 
+        if self.use_ray_spmd_worker:
+            for pp_rank in range(self.parallel_config.pipeline_parallel_size):
+                self.pp_tp_workers.append([])
+                for tp_rank in range(
+                        self.parallel_config.tensor_parallel_size):
+                    # PP=2, TP=4
+                    # pp_tp_workers = [[0, 1, 2, 3], [4, 5, 6, 7]]
+                    rank = (pp_rank * self.parallel_config.tensor_parallel_size
+                            ) + tp_rank
+                    assert len(self.pp_tp_workers[pp_rank]) == tp_rank
+                    assert pp_rank < len(self.pp_tp_workers)
+                    self.pp_tp_workers[pp_rank].append(self.workers[rank])
+
         # This is the list of workers that are rank 0 of each TP group EXCEPT
         # global rank 0. These are the workers that will broadcast to the
         # rest of the workers.
```

Comment:
Possible to somehow integrate this with the previous `self.pp_tp_workers`? If not no worries

Reply:
I attempted twice and it broke either driver mode or adag :( . I think there are pros and cons whether to unify with the same code path. If there is consensus we should go with unification, I can try harder to achieve that.

Reply:
I don't think it's a super huge deal since it's limited to this backend. But if it's possible to do so with a little work I think it cleans up the code a little bit. Can ask for @youkaichao opinion as well

Reply:
Hi @youkaichao, do you have a preference whether we separate the implementations of spmd (aDAG) vs non-spmd or unify them at this point? I had the impression you may perfer keeping them separate for now but wanted to check.